src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1615,45 +1615,45 @@ const char16_t* Compiler::eeGetShortClassName(CORINFO_CLASS_HANDLE clsHnd)
     return param.classNameWidePtr;
 }
 
-const WCHAR* Compiler::eeGetCPString(size_t strHandle)
+void Compiler::eePrintFrozenObjectDescription(const char* prefix, size_t handle)
 {
-#ifdef HOST_UNIX
-    return nullptr;
-#else
-    char buff[512 + sizeof(CORINFO_String)];
-
-    // make this bulletproof, so it works even if we are wrong.
-    if (ReadProcessMemory(GetCurrentProcess(), (void*)strHandle, buff, 4, nullptr) == 0)
+    auto typeOfHandle = *((CORINFO_CLASS_HANDLE*)handle);
```

Comment:
This works for the in-process JIT, but it is broken and can crash for everything else (super PMI, crossgen, nativeaot).

All this logic should be behind JIT/EE interface API as Michal said.